src/middleware/date_query.rs

```diff
@@ -0,0 +1,37 @@
+use chrono::NaiveDate;
+use rocket::request::{FromQuery, Query};
+
+/// Query Parameter for a date range
+///
+/// This param is only parsed, if both a 'from' and a 'to'
+/// query param is set. If one of these dates cannot be parsed
+/// with the format 'YYYY-MM-DD', then no date range can be used.
+///
+#[derive(Copy, Clone, Debug)]
+pub struct DateRangeQueryParam {
+    pub from: NaiveDate,
+    pub to: NaiveDate
+}
+
+impl<'v> FromQuery<'v> for DateRangeQueryParam {
+    type Error = &'static str;
+
+    fn from_query(query: Query<'v>) -> Result<Self, Self::Error> {
```

Comment:
Query is no longer available on master branch.

Reply:
Resolved with FromForm